### PR TITLE
Update controller-runtime, kube-builder, go, alpine and all dependencies, support port-ranges

### PR DESCRIPTION
**Pull request by majst01**
Also add support for EndPort, aka Portranges in CNWP !

TODO:

- [x] manual tests
- [ ] integration tests
- [ ] merge https://github.com/metal-stack/gardener-extension-provider-metal/pull/227 after this was released

with this cwnp:

```yaml
spec:
  ingress:
  - from:
    - cidr: 0.0.0.0/0
    ports:
    - port: 8443
      protocol: TCP
    - port: 8445
      protocol: TCP
    - endPort: 8088
      port: 8080
      protocol: TCP
```

this firewall rule is created:

```
ip saddr { 0.0.0.0-255.255.255.255 } tcp dport { 8080-8088, 8443, 8445 } counter packets 0 bytes 0 accept comment "accept traffic for k8s network policy allow-to-vm-ssh tcp"
```

### CODE REVIEW

File: controllers/droptailer_controller.go [hunk 1]
```diff
@@ -170,28 +166,17 @@ func (r *DroptailerReconciler) SetupWithManager(mgr ctrl.Manager) error {
...
+			{NamespacedName: types.NamespacedName{
+				Name:      "trigger-reconcilation-for-droptailer",
+				Namespace: namespace,
+			}},
+		}
+	})
 
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&corev1.Pod{}, builder.WithPredicates(genericPredicate)).
+		For(&corev1.Pod{}).
```
Comment (majst01) [inline controllers/droptailer_controller.go:179]: unsure if this is expressed correct

**Review by mwennrich — APPROVED** [review]

File: controllers/droptailer_controller.go (hunk 1 above)
Comment (GrigoriyMikhalkin) [reply controllers/droptailer_controller.go:179]: Purpose of predicates is to filter requests. Specifically in this case we filtered out only Generic events. And the rest of events were filtered out by the same criteria(known namespace) in reconcilation logic. So probably this change breaks nothing.

Still, i would suggest to remove this logic from reconcilation and create a predicate for all types of events. As this is recommended approach for filtering requests in controller-runtime.

Comment (majst01) [reply controllers/droptailer_controller.go:179]: OK thanks, how can this be specified ? Link to sample would be much appreciated

Comment (GrigoriyMikhalkin) [reply controllers/droptailer_controller.go]: I pushed changes)

Comment (majst01) [reply controllers/droptailer_controller.go]: Thanks!

**Review by Gerrit91 — APPROVED** [review]

**Pull request merged**